python/lsst/analysis/tools/atools/diffimMetadataMetrics.py

```diff
@@ -0,0 +1,221 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = (
+    "NUnmergedDiaSourcesMetric",
+    "NMergedDiaSourcesMetric",
+    "NGoodPixelsMetric",
+    "NBadPixelsMetric",
+    "NPixelsDetectedPositiveMetric",
+    "NPixelsDetectedNegativeMetric",
+    "NBadPixelsDetectedPositiveMetric",
+    "NBadPixelsDetectedNegativeMetric",
+    "SciencePsfSizeMetric",
+    "TemplatePsfSizeMetric",
+    "ScienceVarianceScaleMetric",
+    "TemplateVarianceScaleMetric",
+    "TemplateCoverageMetric",
+)
+
+from ..actions.scalar import MedianAction
+from ..interfaces import AnalysisTool
+
```

Comment:
There is a lot of duplication here, I'm going to make some suggestions that make use of Analysis_Tools designed configurability.

Start by deleting all of the metrics below and replace with one tool:

```python
# Pick whatever name you want
class DiffimMetadataMetricTool(AnalysisTool):
    """This tool is designed to calculate the median of data produced from diffim metric data"""
    # This is a convenience field to make it easier to set things from the "top level"
    # you could also just reach right in and set things in config as well and leave
    # this out.
    metricKey = Field[str](doc="The metadata field to process, name will be used as the metric name")
    units = Field[str](doc="The units to associate with the metric")

    def setDefaults(self):
        self.process.calculateActions.diffimMedian = MedianAction(vectorKey="")
        self.produce.metric.units = {"diffimMedian": "ct"}

    def finalize(self):
        # This function is implemented to support the convenience fields defined above
        self.process.calculateActions.diffimMedian.vectorKey = self.metricKey
        self.produce.metric.units["diffimMedian"] = self.units
        self.produce.metric.newNames = {"diffimMedian": self.metricKey}
   
```

Alternatively, we could build a tool that takes a configuration of a dict field for all the metrics to produce with one tool

```python
# Pick whatever name you want
class DiffimMetadataMetricTool(AnalysisTool):
    """This tool is designed to calculate the median of data produced from diffim metric data"""
    metrics = DictField[str, str](doc="The metrics to calculate from the task metadata and their units")

    def finalize(self):
        for name, unit in self.metrics.items():
            setattr(self.process.calculateActions, name, MedianAction(vectorKey=name))
        self.produce.metric.units = dict(self.metrics.items())
```

In this case the yaml would change to look something like:
```yaml
atools.diffimMetadataMetric: DiffimMetaDatametricTool
atools.diffimMetadatametric.metrics:
  nMergedDiaSources: ct
  nGoodPixels: ct
  ...
```

This would result in a single atools bundle uploaded to sasquatch with multiple measurements inside it. Both are perfectly reasonable ways to do things, and the dashboards can handle plotting from each the same way. The big difference is that if you are getting these back from the butler, the former will result in n dataset types and n `butler.get` calls. The later will be a single dataset type and a single `butler.get` call with multiple values in it.